src/nni_manager/rest_server/restHandler.ts

```diff
@@ -139,11 +139,18 @@ class NNIRestHandler {
 
     private updateExperimentProfile(router: Router): void {
         router.put('/experiment', expressJoi(ValidationSchemas.UPDATEEXPERIMENT), (req: Request, res: Response) => {
-            this.nniManager.updateExperimentProfile(req.body, req.query.update_type).then(() => {
-                res.send();
-            }).catch((err: Error) => {
-                this.handle_error(err, res);
-            });
+            let experimentMode: string = getExperimentMode();
+            if( experimentMode !== ExperimentStartUpMode.VIEW) {
+                this.nniManager.updateExperimentProfile(req.body, req.query.update_type).then(() => {
+                    res.send();
+                }).catch((err: Error) => {
+                    this.handle_error(err, res);
+                });
+            } else {
+                let message = `Could not update experiment in view mode!`;
```

Comment:
@lvybriage could webui correctly handle this message?

Reply:
Fixed, send 400 code back.